lightning/src/routing/router.rs

```diff
@@ -373,6 +389,19 @@ impl InFlightHtlcs {
 		let mut cumulative_msat = 0;
 		if let Some(tail) = &path.blinded_tail {
 			cumulative_msat += tail.final_value_msat;
+			if tail.hops.len() > 1 {
```

Comment:
I think this method should take into account trampoline hops, might be easy to do here